src/odin/enhancedtrippath.cc

```diff
@@ -508,20 +508,27 @@ bool EnhancedTripLeg_Edge::HasTurnLane(uint16_t turn_lane_direction) const {
 uint16_t EnhancedTripLeg_Edge::ActivateTurnLanesFromLeft(uint16_t turn_lane_direction,
                                                          uint16_t activated_max) {
   uint16_t activated_count = 0;
+
   // Make sure turn lane has a direction
-  if (!HasNonDirectionalTurnLane()) {
-    for (auto& turn_lane : *(mutable_turn_lanes())) {
-      // Stop processing the lanes if the activated maximum has been reached
-      if (activated_count >= activated_max) {
-        break;
-      }
+  if (HasNonDirectionalTurnLane()) {
+    return activated_count;
+  }
 
-      // If the turn lane is in the specified direction then activate the lane
-      // and increment the activated count
-      if (turn_lane.directions_mask() & turn_lane_direction) {
+  for (auto& turn_lane : *(mutable_turn_lanes())) {
+    // Process lanes matching the turn_lane_direction
+    if (turn_lane.directions_mask() & turn_lane_direction) {
+      // TODO: Use lane connectivity to skip impossible lanes
+      // activate upto activated_max lanes
+      if (activated_count < activated_max) {
         turn_lane.set_is_active(true);
         ++activated_count;
       }
+      // Mark non-active lanes as possible
+      turn_lane.set_is_possible(true);
+      // Set the active direction mask in the direction of the maneuver
+      // TODO: Highlighting lane markers for non-active but possible lanes
+      // seems like a questionable thing to do.
```

Comment:
Why questionable? I think you'd have this scenario if you're going straight through an intersection where there's multiple through lanes but one of the through lanes has an option to turn.

Reply:
Because I feel the UI will be a lot more distracting than if the lane direction markers were not highlighted (which is what this attribute is for). Imagine you're turning left, there are 3 lanes going in that direction and you have to be in the middle lane. Having the other lane's direction markers lit up is going to be visually confusing to the driver, and the only way to discern the preferred lane is to rely on the "background" being appropriately highlighted. We should consider making this an opt-in feature, maybe? (we don't have to do it in this PR, but something to consider at a later stage).

(I might be a bit biased here since my mind is trained to expect lanes in a certain way by "other" maps/nav applications 😉).